woocommerce/wc-admin
# AdvancedFilters: Add suggestions to search component from API

Populate a Filter's search component with appropriate suggestions derived from querying API endpoints.

This PR introduces a `<Search />` component to be reused elsewhere using Gutenberg's `FormTokenField`. It accepts a configuration for gathering suggestions from an API endpoint and manipulating the results.

The API work follows a similar pattern as https://github.com/woocommerce/wc-admin/pull/293

## Screenshots

![filter](https://user-images.githubusercontent.com/1922453/44130310-edce3592-a0a0-11e8-8c75-096cd20a546b.gif)

## Testing

1. `/wp-admin/admin.php?page=wc-admin#/analytics/orders`
2. Show > Advanced Filters
3. Add the Products filter
4. Begin typing to add a product and see the suggestions populate
5. See a request being made to gather products
6. Make a selection


Working towards https://github.com/woocommerce/wc-admin/issues/257
